src/main/java/eu/dzhw/fdz/metadatamanagement/projectmanagement/service/DaraService.java

```diff
@@ -344,6 +344,15 @@ private Map<String, Object> getDataForDataPackageTemplate(DataAcquisitionProject
     // Add data for collection mode
     dataForTemplate.put("surveyToCollectionModesMap", computeSurveyToCollectionModesMap(surveys));
 
+    // Add flag for qualitative data in surveys
+    boolean hasQualitativeSurvey = false;
+    for (Survey survey : this.surveyRepository.findByDataAcquisitionProjectId(project.getId())) {
+      if (survey.getDataType().equals(DataTypes.QUALITATIVE_DATA) || survey.getDataType().equals(DataTypes.ALL)) {
```

Comment:
This is where the [error](https://github.com/dzhw/metadatamanagement/actions/runs/8265472205/job/22611857865#step:10:1731) occurs in the pipeline. I misinterpreted the code. The constant `DataTypes.ALL` is a `Set` which apparently is used in model [validation](https://github.com/dzhw/metadatamanagement/blob/ewolf47/issue3241/src/main/java/eu/dzhw/fdz/metadatamanagement/surveymanagement/domain/validation/ValidDataTypeValidator.java#L36) for allowed entries. The class member `Survey::dataType` is a `String` so the second check is not required because it can only be one of the two string constants.